### PR TITLE
Add proposal for events

### PR DESCRIPTION
**Pull request by ptare**
Add events for the sequence a price request goes through.

Compared to earlier versions of this PR, I removed events for `RevealPhaseStarted` and `CommitPhaseStarted`. They may not be as useful because the phases start/stop based on time, not when the first commit or reveal happens.

Signed-off-by: Prasad Tare <prasad.tare@gmail.com>

### CODE REVIEW

File: core/contracts/Voting.sol [hunk 1]
```diff
@@ -636,4 +636,17 @@ contract Voting is Testable, MultiRole, OracleInterface {
...
     }
+
+    event CommitPhaseStarted(uint roundId);
+    event RevealPhaseStarted(uint roundId);
+
+    event VoteCommitted(address voter, bytes32 identifier, uint time, uint roundId);
+    event VoteRevealed(address voter, bytes32 identifier, uint time, uint roundId, int price, uint numTokens);
+    event RewardsRetrieved(address voter, uint rewardsRoundId, uint numTokens);
+
+    event PriceRequestAdded(bytes32 identifier, uint time, uint votingRoundId);
```
Comment (mrice32) [inline core/contracts/Voting.sol]: nit: make `identifier` indexed in this and others?

File: core/contracts/Voting.sol [hunk 2]
```diff
@@ -636,4 +636,17 @@ contract Voting is Testable, MultiRole, OracleInterface {
         // Multiply the total supply at the snapshot by the gatPercentage to get the GAT in number of tokens.
         return snapshottedSupply.mul(gatPercentage);
     }
+
+    event CommitPhaseStarted(uint roundId);
+    event RevealPhaseStarted(uint roundId);
+
+    event VoteCommitted(address voter, bytes32 identifier, uint time, uint roundId);
+    event VoteRevealed(address voter, bytes32 identifier, uint time, uint roundId, int price, uint numTokens);
```
Comment (mrice32) [inline core/contracts/Voting.sol]: nit: maybe make `roundId` indexed in both commit, reveal, and reward retrieval?

File: core/contracts/Voting.sol [hunk 3]
```diff
@@ -636,4 +636,17 @@ contract Voting is Testable, MultiRole, OracleInterface {
         // Multiply the total supply at the snapshot by the gatPercentage to get the GAT in number of tokens.
         return snapshottedSupply.mul(gatPercentage);
     }
+
+    event CommitPhaseStarted(uint roundId);
+    event RevealPhaseStarted(uint roundId);
+
+    event VoteCommitted(address voter, bytes32 identifier, uint time, uint roundId);
```
Comment (mrice32) [inline core/contracts/Voting.sol]: nit: make `voter` indexed here and elsewhere?

**Comment by ptare** [comment]
> Added some suggestions that may be worth discussing - not sure if all of them make sense, just wanted to present some straw men.

Making these fields indexed seems reasonable to me.

**Comment by ptare** [comment]
> > Added some suggestions that may be worth discussing - not sure if all of them make sense, just wanted to present some straw men.
> 
> Making these fields indexed seems reasonable to me.

PTAL.

**Comment by mrice32** [comment]
Agreed that the commit and reveal phase events might not make as much sense. We can add them later if we deem them necessary.

**Comment by ptare** [comment]
Issue #481

**Pull request merged**